### PR TITLE
Implement list command

### PR DESCRIPTION
**Pull request by yokaze**
This PR contains the following four commits:

### 1. Respect cached files

Fix Makefile and Helm file cache to make them work properly.

### 2. Run test pods

Deploy some test pods with L3 policies.
- `self`: entry point
- `ingress-explicit-allow`: egress from `self` and ingress to `ingress-explicit-allow` are allowed
- `ingress-no-rule`: egress from `self` is allowed and ingress to `ingress-no-rule` is implicitly allowed (no rule)
- `ingress-explicit-deny`: egress from `self` is allowed but ingress to `ingress-explicit-deny` is explicitly denied
- `egress-implicit-deny`: egress from `self` to `egress-implicit-deny` is implicitly denied
- `egress-explicit-deny`: egress from `self` to `egress-explicit-deny` is explicitly denied

### 3. Implement list command

Implement `cilium-policy list` that shows the list of NetworkPolicy applied to a specific pod.
Currently the available output format is only `json`: `simple` will be available in future.

Also `cilium-policy dump` is heavily refactored using common helper functions.

### 4. Check go.mod
Just check `go.mod` is up-to-date.

Signed-off-by: Daichi Sakaue <daichi-sakaue@cybozu.co.jp>

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -5,7 +5,7 @@ on:
     branches:
       - main
 env:
-  cache-version: 1
+  cache-version: "2024-04-05"
```
Comment (chez-shanpu) [inline .github/workflows/ci.yaml:8]: Why do you use date as cache-version?
And also, I wonder if we need cache-version. Are there any cases that `hashFiles` is not sufficient?

File: Makefile [hunk 2]
```diff
@@ -26,39 +30,48 @@ help: ## Display this help
...
-setup: $(HELM) $(JQ) $(KUBECTL) $(KUSTOMIZE) $(YQ) ## Install necessary tools
-	GOBIN=$(TOOLS_DIR) go install sigs.k8s.io/kind@v$(KIND_VERSION)
-	GOBIN=$(TOOLS_DIR) go install honnef.co/go/tools/cmd/staticcheck@latest
-	GOBIN=$(TOOLS_DIR) go install github.com/cybozu-go/golang-custom-analyzer/cmd/custom-checker@latest
+setup: $(CUSTOMCHECKER) $(HELM_BINARY) $(KIND) $(JQ) $(KUBECTL) $(KUSTOMIZE) $(STATICCHECK) $(YQ) ## Install necessary tools
 	$(HELM) repo add cilium https://helm.cilium.io/
 	$(HELM) repo update cilium
 
-$(TOOLS_DIR):
-	mkdir -p $(TOOLS_DIR)
```
Comment (chez-shanpu) [inline Makefile:37]: Why did you remove `$(TOOLS_DIR)` target?

Comment (yokaze) [reply Makefile:37]: @chez-shanpu 
It enforces us to download tools in the directory every time.
For example, `$(JQ)` always has a timestamp older than that of `$(TOOLS_DIR)`.
As a result, `make` assumes `$(JQ)` to be stale and it needs update.

Comment (chez-shanpu) [reply Makefile:37]: I see. Thank you!

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (yokaze) [reply .github/workflows/ci.yaml:8]: @chez-shanpu 
It is an old practice to allow invalidating caches when it is contaminated by a broken CI workflow.
However I always hesitate to update the version one: it looks like having a special meaning. Using a day looks more meaningless.

Comment (chez-shanpu) [reply .github/workflows/ci.yaml:8]: @yokaze I see. How about deleting a cache like `gh cache delete` when it is contaminated instead of using `cache-version`?

Comment (yokaze) [reply .github/workflows/ci.yaml]: @chez-shanpu 
It looks reasonable, I removed the `cache-version` parameter. Thank you for the info!

**Review by chez-shanpu — APPROVED** [review]
LGTM
